std/algorithm/comparison.d

```diff
@@ -699,6 +699,9 @@ int cmp(alias pred = "a < b", R1, R2)(R1 r1, R2 r2) if (isSomeString!R1 && isSom
     assert(result > 0);
 }
 
+enum areEquable(T, U) = is(typeof({ return T.init == U.init; })); // TODO move to std.traits
+enum haveEquableLengths(T, U) = is(typeof({ return T.init.length == U.init.length; })); // TODO move to std.traits
```

Comment:
I guess the end result was size_t as I recall debates over BitArray potentially having ulong length on 32-bit systems. It never made it and we all agreed size_t is the way to go IRC.